crates/node-builder/src/builder.rs

```diff
@@ -440,6 +441,13 @@ where
         // Does not do anything on windows.
         fdlimit::raise_fd_limit()?;
 
+        // Limit the global rayon thread pool, reserving 2 cores for the rest of the system
+        ThreadPoolBuilder::new()
+            .num_threads(
+                available_parallelism().map_or(25, |cpus| max(cpus.get().saturating_sub(2), 2)),
+            )
+            .build_global()?;
```

Comment:
can we ignore the error here and log it instead?
in case the user already configured this